src/scalar/mod.rs

```diff
@@ -34,6 +41,8 @@ pub fn is_scalar(type_name: &str) -> bool {
 pub enum CustomScalar {
```

Comment:
@coderabbitai please suggest some documentation for why we need `CustomScalar` as far as I know, its to generate documentation.